composer/trainer/trainer.py

```diff
@@ -1018,9 +1146,150 @@ def saved_checkpoints(self) -> List[Tuple[Timestamp, List[pathlib.Path]]]:
             return []
         return self._checkpoint_saver.saved_checkpoints
 
-    def fit(self):
-        """Train and evaluate the model on the provided data."""
-        # Print any exception, so it can be caputred by any callbacks or loggers (e.g. WandB, FileLogger)
+    def fit(
+        self,
+        *,
+        # Train Dataloader
+        train_dataloader: Optional[Union[Iterable, DataSpec, Dict[str, Any]]] = None,
+        train_dataloader_label: str = "train",
+        train_subset_num_batches: Optional[int] = None,
+        compute_training_metrics: Optional[bool] = None,
```

Comment:
this is barely used, I think we can remove this option?

Reply:
This was originally added because researchers requested it - we should prob confirm with people first

Reply:
Going to leave this here for now, but this will be addressed in metrics API refactor.

I suspect this parameter would be renamed to something like `train_metrics: Optional[List[str]] = None` where the user specifies the metric names to compute on the train dataset.